src/py/flwr/cli/config_utils.py

```diff
@@ -112,8 +113,11 @@ def _validate_run_config(config_dict: Dict[str, Any], errors: List[str]) -> None
     for key, value in config_dict.items():
         if isinstance(value, dict):
             _validate_run_config(config_dict[key], errors)
-        elif not isinstance(value, str):
-            errors.append(f"Config value of key {key} is not of type `str`.")
+        elif not isinstance(value, get_args(UserConfigValue)):
+            raise ValueError(
+                f"The value for key {key} needs to be of type `int`, `float`, "
+                "`bool, `str`, or  a `dict` of those.",
```

Comment:
Wouldn't `dict` be an error at this point?
```suggestion
                f"The value for key {key} needs to be of type `bool`, `float`, `int` "
                "or `str`.",
```